### PR TITLE
Add exchange unit testing.

### PR DESCRIPTION
**Pull request by wlynch**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,12 +12,14 @@ require (
 	github.com/chainguard-dev/terraform-infra-common v0.6.85
 	github.com/cloudevents/sdk-go/v2 v2.15.2
 	github.com/coreos/go-oidc/v3 v3.11.0
+	github.com/go-jose/go-jose/v3 v3.0.3
```
Comment (cpanato) [inline go.mod:15]: can we use ` "github.com/go-jose/go-jose/v4" `

**Review by cpanato — CHANGES_REQUESTED** [review]
looks cool, jsut a request

File: go.mod (hunk 1 above)
Comment (wlynch) [reply go.mod]: Done!

**Review by cpanato — APPROVED** [review]
thank you

**Pull request merged**